extensions/amp-story/0.1/amp-story-auto-ads.js

```diff
@@ -117,7 +120,7 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
     Services.extensionsFor(this.win)./*OK*/installExtensionForDoc(
         ampdoc, MUSTACHE_TAG);
 
-    ampStoryElement.getImpl().then(impl => {
+    this.storyImplPromise_ = ampStoryElement.getImpl().then(impl => {
```

Comment:
you might just return this promise, it will block the lifecycle of the element.
but please test it out if i am right

Reply:
You are right. changed.